### ISSUE TITLE
thread 'main' panicked at 'lib/src/network/https_openssl.rs:306 sozu_lib::network::https_openssl

### ISSUE THREAD

**Issue by BlackYoup** [sozu-proxy/sozu#353]
I don't know how to reproduce but it crashed a few times in a row.

```
2018-02-17T14:24:31Z 3813079583549288 4163 WRK-08 TRACE sozu_lib::network::protocol::pipe       pipe readable
2018-02-17T14:24:31Z 3813079583618028 4163 WRK-08 DEBUG sozu_lib::network::protocol::pipe       5cec339e-efc6-482c-b539-51a743823f3e    unknown         FRONT [Some(Token(739))]: read 0 bytes
2018-02-17T14:24:31Z 3813079596339722 4163 WRK-08 ERROR 5cec339e-efc6-482c-b539-51a743823f3e    unknown         [Some(Token(739))] front socket error, closing the connection
2018-02-17T14:24:31Z 3813079596385542 4163 WRK-08 TRACE sozu_lib::network::https_openssl        front readable  interpreting client order CloseClient
thread 'main' panicked at 'lib/src/network/https_openssl.rs:306 sozu_lib::network::https_openssl this should not have panicked:
tried to unwrap `None` in:
unwrap_msg!(self.http())', lib/src/util.rs:18:18
```

**Comment by BlackYoup** [comment]
I thought I had a different panic but it seems to be the same, maybe not triggered the same way though.

```
2018-02-17T14:26:58Z 3813226512522374 25203 WRK-09 INFO b5cf9257-d5de-450c-aae5-2d72272dd3a3 MyApp  client-ip:58462 -> 127.0.0.1:8080     72052 72049 476 237     200 OK my-domain.com GET my-url/?test
2018-02-17T14:26:58Z 3813226512553684 25203 WRK-09 DEBUG        sozu_lib::network::protocol::http       b5cf9257-d5de-450c-aae5-2d72272dd3a3 MyApp   keep alive front/back
2018-02-17T14:26:58Z 3813226512593723 25203 WRK-09 TRACE        sozu_lib::network::https_openssl        front writable  interpreting client order Continue
2018-02-17T14:26:58Z 3813226512743192 25203 WRK-09 DEBUG        sozu_lib::network::protocol::http       ed40e1a1-05b6-445a-a16c-e6737ede505a MyApp          FRONT: read 509 bytes
2018-02-17T14:26:58Z 3813226512772211 25203 WRK-09 ERROR        ed40e1a1-05b6-445a-a16c-e6737ede505a MyApp  78.221.4.55:56540 -> 127.0.0.1:8080     0 0 509 0       - - - | front socket error, closing the connection
2018-02-17T14:26:58Z 3813226512794991 25203 WRK-09 TRACE        sozu_lib::network::https_openssl        front readable  interpreting client order CloseClient
2018-02-17T14:26:58Z 3813226512840410 25203 WRK-09 DEBUG        sozu_lib::network::protocol::http       ed40e1a1-05b6-445a-a16c-e6737ede505a MyApp          PROXY [4 -> 5] CLOSED BACKEND
2018-02-17T14:26:58Z 3813226512965929 25203 WRK-09 DEBUG        sozu_lib::network::protocol::http       f2ceada6-e3ea-4e95-a587-b65f55e647cf MyApp          FRONT: read 925 bytes
2018-02-17T14:26:58Z 3813226512992099 25203 WRK-09 ERROR        f2ceada6-e3ea-4e95-a587-b65f55e647cf MyApp  78.221.4.55:58462 -> 127.0.0.1:8080     0 0 925 0       - - - | front socket error, closing the connection
2018-02-17T14:26:58Z 3813226513027878 25203 WRK-09 TRACE        sozu_lib::network::https_openssl        front readable  interpreting client order CloseClient
2018-02-17T14:26:58Z 3813226513063248 25203 WRK-09 DEBUG        sozu_lib::network::protocol::http       f2ceada6-e3ea-4e95-a587-b65f55e647cf MyApp          PROXY [9 -> 10] CLOSED BACKEND
2018-02-17T14:26:58Z 3813226513501713 25203 WRK-09 TRACE        sozu_lib::network::https_openssl        ref: Ssl { state: "SSLv3 read client hello C" }
2018-02-17T14:26:58Z 3813226513529573 25203 WRK-09 DEBUG        sozu_lib::network::https_openssl        checking servername: my-domain.com
2018-02-17T14:26:58Z 3813226513542662 25203 WRK-09 DEBUG        sozu_lib::network::https_openssl        looking for fingerprint for "my-domain.com"
2018-02-17T14:26:58Z 3813226513558842 25203 WRK-09 DEBUG        sozu_lib::network::https_openssl        looking for context for "my-domain.com" with fingerprint CertFingerprint(my-fingerprint)
2018-02-17T14:26:58Z 3813226513579522 25203 WRK-09 DEBUG        sozu_lib::network::https_openssl        found context for "my-domain.com"
2018-02-17T14:26:58Z 3813226513599292 25203 WRK-09 DEBUG        sozu_lib::network::https_openssl        servername is now Some("my-domain.com")
2018-02-17T14:26:58Z 3813226513781570 25203 WRK-09 ERROR        accept: handshake failed: MidHandshakeSslStream { stream: SslStream { stream: TcpStream { sys: TcpStream { inner: TcpStream { addr: V4(sozu-ip:443), peer: V4(client-ip:56546), fd: 15 } }, selector_id: SelectorId { id: AtomicUsize(0) } }, ssl: Ssl { state: "SSLv3 read finished A" } }, error: Stream(Error { repr:
 Custom(Custom { kind: ConnectionAborted, error: StringError("unexpected EOF observed") }) }) }
2018-02-17T21:26:58Z 3813226513867529 25203 WRK-09 TRACE        sozu_lib::network::https_openssl        front readable  interpreting client order CloseClient
thread 'main' panicked at 'lib/src/network/https_openssl.rs:306 sozu_lib::network::https_openssl this should not have panicked:
tried to unwrap `None` in:
unwrap_msg!(self.http())', lib/src/util.rs:18:18```

**Comment by Geal** [comment]
should be fixed by e51bfcf57044967f5

**Comment by BlackYoup** [comment]
I couldn't reproduce it since the commit, I'll mark it as close and re-open if I see it again
